rfc/rfc-0009-supply-chain-switches-template-on-flag.md

```diff
@@ -1,3 +1,197 @@
+  selector:
+    app: web
+  resources:
+    - name: source-provider
+      templateRef:
+        kind: ClusterSourceTemplate
+        options:                                  # <--- a list
+        - name: git-template
+          selector:
+            matchFields:
+              - key: workload.spec.source.git       # <--- path to field in template context
+                operator: Exists
+        - name: imgpkg-bundle-template
+          selector:
+            matchFields:
+              - key: workload.spec.source.image       # <--- path to field in template context
+                operator: Exists
+
+---
+apiVersion: kontinue.io/v1alpha1
+kind: Workload
+metadata:
+  name: petclinic
+  labels:
+    app: web
+spec:
+  source:
+    git:                                        # <--- Presence of this field will determine the conditional
+      url: https://github.com/spring-projects/spring-petclinic.git
+      ref:
+        branch: main
+```
+
+1. As labels are present in the workload metadata, any behavior possible using just label selectors
+  will be possible using the entire workload.
+
+### Error conditions
+
+Exactly one template should be chosen at each step. If no template is chosen, or multiple templates match, an error
+should be thrown.
+
+### Matching syntax
+
+Kubernetes resources "such as Job, Deployment, ReplicaSet, and DaemonSet, support set-based requirements". The
+syntax for matchExpressions allows for asserting `In` or `NotIn` a specified array of values, or to assert
+that a key `Exists` or `DoesNotExist`. matchExpressions is a list of assertions, which are ANDed.
+[link](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#resources-that-support-set-based-requirements)
+
+## Notes
+
+1. While this RFC has referred to workloads and supply chains throughout, the proposal is to apply this
+  behavior to Delivery/Deliverable as well.
+2. This RFC benefited from comments in the pull request, discussions of the Cartographer team and community
+  and ideas in [this proposal](https://gist.github.com/squeedee/723be000c4f2ee40ce4c9ac020cbf4fc) 
+  from [Rasheed Abdul-Aziz] (https://github.com/squeedee)
+
+## Concerns
+
+Deterministic supply chains are desirable. That is, given a workload it is valuable to be able to
+state which supply chain will be chosen and what the path through that supply chain will be.
+
+However, it is not certain that Cartographer will have the luxury of determinism. It is possible that the
+output of a choreographed resource may be important to the next step of a supply chain. Essentially, what if
+an object emits some values that can be handled by controller A, and other objects that can only be handled
+by controller B? There would need to be a step which could switch on the emitted values. This RFC does not
+handle this case. An RFC to use the entire templating context for template choice would address this use case.
+
+## Alternatives
+
+### Label selectors for choosing templates
```

Comment:
I might be missing something, so please correct me if I'm wrong, but it feels like this approach could actually be a separate rule for `options`. e.g., one could make use of both proposals with something like

```yaml
  resources:
    - name: source-provider
      templateRef:
        kind: ClusterSourceTemplate
        options:                                  # <--- a list
        - name: git-template
          selector:
            matchFields:
              - key: workload.spec.source.git       # <--- path to field in template    context
                operator: Exists
        - name: imgpkg-bundle-template
          selector:
            matchLabels:
              app.tanzu.vmware.com/source-type: imgpkg-bundle
```